### PR TITLE
Workaround JDK-8311164: CPU_HT is set randomly on hybrid CPUs like Alder Lake

### PR DESCRIPTION
**Pull request by jankratochvil**
SSIA

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Anton Kozlov](https://openjdk.org/census#akozlov) (@AntonKozlov - **Reviewer**)
 * [Radim Vansa](https://openjdk.org/census#rvansa) (@rvansa - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/crac.git pull/89/head:pull/89` \
`$ git checkout pull/89`

Update a local copy of the PR: \
`$ git checkout pull/89` \
`$ git pull https://git.openjdk.org/crac.git pull/89/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 89`

View PR using the GUI difftool: \
`$ git pr show -t 89`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/crac/pull/89.diff">https://git.openjdk.org/crac/pull/89.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/crac/pull/89#issuecomment-1614457379)

### CODE REVIEW

File: src/hotspot/cpu/x86/vm_version_x86.cpp [hunk 1]
```diff
@@ -2588,6 +2592,10 @@ void VM_Version::initialize() {
   uint64_t   CPUFeatures_x64 = CPUFeatures_parse(GLIBCFeatures_x64);
   uint64_t       features_missing =   CPUFeatures_x64 & ~      _features;
   uint64_t glibc_features_missing = GLIBCFeatures_x64 & ~_glibc_features;
+
+  // Workaround JDK-8311164: CPU_HT is set randomly on hybrid CPUs like Alder Lake.
+  features_missing &= ~CPU_HT;
```
Comment (AntonKozlov) [inline src/hotspot/cpu/x86/vm_version_x86.cpp:2597]: Note: I just realized these missing features are different from the others, it's more "extra features" -- they are specified via the option, but not available on the current CPU.

**Review by AntonKozlov — APPROVED** [review]

File: src/hotspot/cpu/x86/vm_version_x86.cpp (hunk 1 above)
Comment (jankratochvil) [reply src/hotspot/cpu/x86/vm_version_x86.cpp:2597]: I am not sure what you mean here. This part of the code checks that the `-XX:CPUFeatures=0x...,0x...` specifies only a subset of the current options, never superset. Or do you have a suspection there exist some other options besides `CPU_HT` which are not fatal to use even if the CPU does not support them?

**Review by rvansa — APPROVED** [review]
The testsuite passes on my Alder Lake (i7-12700H)

File: src/hotspot/cpu/x86/vm_version_x86.cpp (hunk 1 above)
Comment (AntonKozlov) [reply src/hotspot/cpu/x86/vm_version_x86.cpp:2597]: > This part of the code checks that the -XX:CPUFeatures=0x...,0x... specifies only a subset of the current options, never superset.

Good, that was my understanding as well. I mean just the variables names are the same, but the meaning is different. But I don't suppose it should be fixed. The PR is good.

Thanks!

**Pull request closed without merging**